server/odc-service/src/test/java/com/oceanbase/odc/service/sqlcheck/MySQLCheckerTest.java

```diff
@@ -1362,6 +1364,211 @@ public void check_wrongPlExists_violationGenerated() {
+                .thenReturn(resultSet);
+        DefaultSqlChecker updateChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualUpdate = updateChecker.check(update);
+        Assert.assertEquals(0, actualUpdate.size());
+    }
+
+    @Test
+    public void check_restrictSqlAffectedRows4OBDelete() {
+        String delete =
+                "delete from users where id in ('1', '2', '3')";
+        List<String> resultSet = Arrays.asList(
+                "==================================================",
+                "|ID|OPERATOR          |NAME|EST.ROWS|EST.TIME(us)|",
+                "--------------------------------------------------",
+                "|0 |DISTRIBUTED INSERT|    |0       |20          |",
+                "|1 |└─EXPRESSION      |    |3       |1           |",
+                "==================================================");
+
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(resultSet);
+        DefaultSqlChecker deleteChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualDelete = deleteChecker.check(delete);
+        Assert.assertEquals(1, actualDelete.size());
+    }
+
+    @Test
+    public void check_restrictSqlAffectedRows4OBOthers() {
+        String select =
+                "select id, name, age from users where id in ('1', '2')";
+        List<String> resultSet = Arrays.asList(
+                "==================================================",
+                "|ID|OPERATOR          |NAME|EST.ROWS|EST.TIME(us)|",
+                "--------------------------------------------------",
+                "|0 |DISTRIBUTED INSERT|    |4       |20          |",
+                "|1 |└─EXPRESSION      |    |4       |1           |",
+                "==================================================");
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(resultSet);
+        DefaultSqlChecker selectChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualSelect = selectChecker.check(select);
+        Assert.assertEquals(0, actualSelect.size());
+    }
+
+    @Rule
+    public final ExpectedException exceptionRule = ExpectedException.none();
```

Comment:
The field is supposed to be put in first place in the class.

Reply:
That's true, but what I'm considering is that it may be easy to understand on the actual case. It's not a big problem. I put it at the top.